src/main/java/org/springframework/data/elasticsearch/core/SearchOperations.java

```diff
@@ -170,32 +202,203 @@ default long count(Query query, IndexCoordinates index) {
+	 * @param index the index to run the query against
+	 * @return a page with aggregations
+	 */
+	<T> AggregatedPage<SearchHit<T>> searchForPage(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi-search against elasticsearch and return result as {@link List} of {@link Page}
+	 *
+	 * @param queries the queries
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of pages with the results
+	 */
+	<T> List<Page<SearchHit<T>>> searchForPage(List<? extends Query> queries, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi-search against elasticsearch and return result as {@link List} of {@link Page}
+	 *
+	 * @param queries the queries
+	 * @param classes the entity classes used for the queries
+	 * @param index the index to run the query against
+	 * @return list of pages with the results
+	 */
+	List<AggregatedPage<? extends SearchHit<?>>> searchForPage(List<? extends Query> queries, List<Class<?>> classes,
+			IndexCoordinates index);
+
+	/**
+	 * Execute the criteria query against elasticsearch and return result as {@link List}
+	 *
+	 * @param <T> element return type
+	 * @param query the query to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of found objects
+	 */
+	<T> SearchHits<T> searchForList(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Execute the multi search query against elasticsearch and return result as {@link List}
+	 *
+	 * @param queries the queries to execute
+	 * @param clazz the entity clazz used for property mapping
+	 * @param index the index to run the query against
+	 * @param <T> element return type
+	 * @return list of found objects
+	 */
+	default <T> List<List<SearchHit<T>>> searchForList(List<Query> queries, Class<T> clazz, IndexCoordinates index) {
+		return searchForPage(queries, clazz, index).stream().map(Page::getContent).collect(Collectors.toList());
+	}
+
+	/**
+	 * Execute the multi search query against elasticsearch and return result as {@link List}
+	 *
+	 * @param queries the queries to execute
+	 * @param classes the entity classes used for property mapping
+	 * @param index the index to run the query against
+	 * @return list of list of found objects
+	 */
+	default List<List<?>> searchForList(List<Query> queries, List<Class<?>> classes, IndexCoordinates index) {
+		return searchForPage(queries, classes, index).stream().map(Page::getContent).collect(Collectors.toList());
+	}
+
+	/**
+	 * Execute the query against elasticsearch and return ids
+	 *
 	 * @param query the query to execute
 	 * @param clazz the entity clazz used for property mapping
 	 * @param index the index to run the query against
+	 * @return list of found object ids
+	 */
+	List<String> searchForIds(Query query, Class<?> clazz, IndexCoordinates index);
```

Comment:
Let's not introduce this method as this method creates API surface for something that is easily consumable with `SearchHit.getId()`. Internally, there's no mapping happening and when using `SearchDocument`, we also skip mapping.